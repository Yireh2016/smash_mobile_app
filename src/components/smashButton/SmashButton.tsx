import * as React from 'react';
import {
  StyleProp,
  ViewStyle,
  View,
  StyleSheet,
  TouchableOpacityProperties,
  TouchableOpacity,
} from 'react-native';
import { H2 } from '../texts/Texts';
import useMyTheme from '../../hooks/useMyTheme';

interface SmashButtonProps {
  label: string;
  onPress?(): void;
  props?: TouchableOpacityProperties;
  style?: StyleProp<ViewStyle>;
  ref?: any;
  disabled?: boolean;
  type?: string;
  testID?: string;
}

const SmashButton: React.FunctionComponent<SmashButtonProps> = ({
  onPress,
  label,
  style,
  props,
  disabled = false,
  type = 'primary',
  testID,
}) => {
  const theme = useMyTheme();
  const styles = StyleSheet.create({
    btnContainer: {
      backgroundColor: type?.match('secondary')
        ? theme.palette.primary
        : theme.palette.secondary,
      borderRadius: 12,
      paddingTop: 10,
      paddingBottom: 10,
      width: '100%',
      borderWidth: 1,
      borderColor: type?.match('secondary')
        ? theme.palette.secondary
        : 'transparent',
    },
    btnDisabled: {
      backgroundColor: theme.palette.gray,
      borderRadius: 12,
      paddingTop: 10,
      paddingBottom: 10,
      width: '100%',
      borderWidth: 1,
      borderColor: theme.palette.white,
    },
    btnLabel: {
      color: disabled ? theme.palette.black : theme.palette.white,
      textAlign: 'center',
    },
  });
  return (
    <View style={disabled ? styles.btnDisabled : styles.btnContainer}>
      {type.match('no-action') ? (
        <H2 style={styles.btnLabel}>{label}</H2>
      ) : (
        <TouchableOpacity
          testID={testID}
          disabled={disabled}
          style={style}
          {...props}
          onPress={onPress}
        >
          <H2 style={styles.btnLabel}>{label}</H2>
        </TouchableOpacity>
      )}
    </View>
  );
};

export default SmashButton;
