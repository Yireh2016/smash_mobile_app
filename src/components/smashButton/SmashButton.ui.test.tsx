import * as React from 'react';
import renderer from 'react-test-renderer';
import SmashButton from './SmashButton';

describe('SmashButton renders correctly', () => {
  it('SmashButton primary renders correctly', () => {
    const ComponentProps = {
      label: 'test',
      onPress: () => {},
    };
    const Element = () => {
      return <SmashButton {...ComponentProps} />;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('SmashButton secondary renders correctly', () => {
    const ComponentProps = {
      label: 'test',
      onPress: () => {},
      type: 'secondary',
    };
    const Element = () => {
      return <SmashButton {...ComponentProps} />;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
