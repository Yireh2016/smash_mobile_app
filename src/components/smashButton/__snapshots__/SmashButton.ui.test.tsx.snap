// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmashButton renders correctly SmashButton primary renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#29b782",
      "borderColor": "transparent",
      "borderRadius": 12,
      "borderWidth": 1,
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`SmashButton renders correctly SmashButton secondary renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#10202b",
      "borderColor": "#29b782",
      "borderRadius": 12,
      "borderWidth": 1,
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      test
    </Text>
  </View>
</View>
`;
